scripts/west_commands/runners/generic_command.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2017 Linaro Limited.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Runner for flashing with external command option.'''
+
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+
+class GenericCommandRunner(ZephyrBinaryRunner):
+    # This is a special flash command which uses the given
+    # 'command' option to achieve the goal copying test
+    # case image and grub boot image to board
```

Comment:
This is still mixing up things that are specific to your board into a runner that has nothing to do with it:

- this comment which mentions grub
- the git shortlog which says the area is "arch/x86" despite only editing files in "scripts/west_commands"
- assuming that something has created a `zephyr.strip` file which must be passed to the command

It also doesn't include the suggestion that the user should be able to pass arbitrary commands at the command line.

I think the best way to get this PR merged is if I provide you with some infrastructure which you can use for your board. Let's start here if we can: https://github.com/zephyrproject-rtos/zephyr/pull/20273

If we can agree on the infrastructure you need for up_squared and merge that, then we can come back to this PR with a commit that just touches the up_squared board.cmake.

Reply:
it's great.

Reply:
touched a board.cmake file to set flash runner for up squared according to #20273, is it ok?